library/common/jni/jni_utility.cc

```diff
@@ -109,6 +113,11 @@ jobject native_map_to_map(JNIEnv* env, envoy_map map) {
 }
 
 envoy_data buffer_to_native_data(JNIEnv* env, jobject j_data) {
+  size_t data_length = static_cast<size_t>(env->GetDirectBufferCapacity(j_data));
```

Comment:
@carloseltuerto I believe we need to fix this. I am seeing malloc aborts thanks to data_length being obscenely high. For instance on a debug branch of the Lyft app I am seeing cases where the returned value here is `18446744073709551615`

cc @goaway

Reply:
We also need better testing in these utility functions. They influence a lot of other behavior. So even if the larger feature in this PR was opt in, changing the utility functions in the bridge layer has a wider impact.

Reply:
So stating the obvious here but 18446744073709551615 is 2^64-1 which is `-1`. I think the bug here is that for some reason `GetDirectBufferCapacity` is returning `-1`? Only reason I can think that could happen is either `j_data` is not an initialized buffer or the JVM on that case for some reason doesn't support JNI access to buffers? The latter seems strange. Just theorizing.

Reply:
I'm betting this is happening when the buffer is not a direct buffer.

Reply:
Pretty sure I see the issue, will post a fix shortly.